### ISSUE TITLE
Revisit the training loss for various loss function

### ISSUE THREAD

**Issue by minseok-ryu** [APPFL/APPFL#84]
Need to revisit the way of producing training loss for various loss functions
